querybook/webapp/ui/Comment/Comments.tsx

```diff
@@ -0,0 +1,257 @@
+        />
+    );
+
+    const handleOpenThread = React.useCallback((threadId: number) => {
+        setOpenThreadIds((curr) => new Set([...curr, threadId]));
+    }, []);
+
+    const renderThreadCommentDOM = (
+        parentCommentId: number,
+        comments: IComment[]
+    ) => {
+        if (openThreadIds.has(parentCommentId)) {
+            return comments.map(renderFlatCommentDOM);
+        }
+        const uids = Array.from(
+            new Set(comments.map((comment) => comment.uid))
+        );
+        return (
+            <div className="ClosedCommentThread flex-row">
+                <div className="flex-row mr8">
+                    {uids.map((uid) => (
+                        <UserAvatar
+                            key={`thread-avatar-${uid}-${parentCommentId}`}
+                            uid={uid}
+                            tiny
+                        />
+                    ))}
+                </div>
+                <StyledText
+                    className="ThreadCount mr8"
+                    size="xsmall"
+                    color="accent"
+                    cursor="default"
+                >
+                    {comments.length}{' '}
+                    {comments.length === 1 ? 'Reply' : 'Replies'}
+                </StyledText>
+                <span className="LastReplyDate">
+                    <StyledText size="xsmall" color="lightest" cursor="default">
+                        Last reply{' '}
+                        {fromNow(comments[comments.length - 1].updated_at)}
+                    </StyledText>
+                </span>
+                <span className="HoverText">
+                    <StyledText size="xsmall" color="lightest" cursor="default">
+                        View Thread
+                    </StyledText>
+                </span>
+            </div>
+        );
+    };
+
+    const renderCommentDOM = () =>
+        comments.map((comment: IComment) => {
+            if (comment.child_comments) {
+                return (
+                    <>
+                        {renderFlatCommentDOM(comment)}
+                        <div
+                            className="CommentThread mt12"
+                            onClick={() => handleOpenThread(comment.id)}
+                        >
+                            {renderThreadCommentDOM(
+                                comment.id,
+                                comment.child_comments
+                            )}
+                        </div>
+                    </>
+                );
+            }
+            return renderFlatCommentDOM(comment);
+        });
+
+    const renderEditingCommentWarning = () => (
+        <div className="Comments-edit-warning flex-row ml48 mb4">
+            <StyledText
+                size="xsmall"
+                color="lightest"
+                cursor="default"
+                className="mr16"
+            >
+                Editing Comment
```

Comment:
would be cool to say:

Editing Comment: (first few words of the comment being edited)...

or 

we can highlight the comment being edited as well